scripts/flytekit_sagemaker_runner.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import absolute_import
+
+import argparse
+import logging
+import subprocess
+from os import environ
+
+FLYTE_ARG_PREFIX = "--__FLYTE"
+FLYTE_ENV_VAR_PREFIX = f"{FLYTE_ARG_PREFIX}_ENV_VAR_"
+FLYTE_CMD_PREFIX = f"{FLYTE_ARG_PREFIX}_CMD_"
+FLYTE_ARG_SUFFIX = "__"
+
+# An example for a valid command:
```

Comment:
Please take a look at this commit @bnsblue

Reply:
@EngHabu This is not entirely correct. We currently support the following format
```
--__FLYTE_SAGEMAKER_CMD__ service_venv+pyflyte-execute
--__FLYTE_CMD_ARG_arg1__ value1 --__FLYTE_CMD_ARG_arg2__ value2 --__FLYTE_CMD_ARG_arg3__ value3 --__FLYTE_CMD_ARG_flag__ --__FLYTE_ENV_VAR_env_var1__ envvalue1 --__FLYTE_ENV_VAR_env_var2__ envvalue2 
```

Reply:
The reason for doing this is that in a end2end2 flow, the variable name and values will be passed to the runner script via the Hyperparameter field of the SM CRD. However, we found that there's a limit on the size of hyperparameter values (and I believe the's also a limit on the size of the hyperparameter name). Using the format that we currently support, we are trying to minimize both the size of the variable name and the size of the value.

Reply:
see this as well: https://github.com/lyft/flytekit/pull/156#discussion_r479418016

Reply:
It's not necessary... just borrowing from k8s & docker concepts... there are commands and there are args...
I'm now assuming you will pass `service_venv` as well as `pyflyte-execute` as just regular __FLYTE_CMD_* params... nothing is special about these two..

and the reason for that is, I don't like [this](https://github.com/lyft/flyteplugins/pull/113/files#diff-d8c750da896887beb6e43f80268f2351R286). I think the plugin should not make any assumptions about the command and how it looks like... so I would rather keep the "protocol" simple between the two to say:
- The SDK will write a container proto with cmd & args
- The plugin will pass all the CMDs and Args as `__FLYTE_CMD_*`
- The SDK must then reconstruct the original command from those...

Remember that FlyteKit might not even be in the container the plugin will launch... so keeping this protocol clear and easy to implement (in a bash script) is just as important...

Let me make the env var format work... that's the only difference, right?

Reply:
see this: https://github.com/lyft/flytekit/pull/156#discussion_r479428335